src/app/api/v1/auth/verify/route.tsx

```diff
@@ -1,11 +1,68 @@
 import { NextResponse, type NextRequest } from "next/server";
+import { SiweMessage } from "siwe";
+import {
+  generateJwt,
+  getScopeForUser,
+  getExpiryForUser,
+} from "@/app/lib/auth/serverAuth";
 
+// This should create a user id record if there is not one already
+// This should not generate an API key
 export async function POST(request: Request) {
+  const body = await request.json();
+  const { message, signature, nonce } = body;
+
+  // N.B. the below assumes the address we're validating is an
+  // EOA. For EIP-1271, a provider will be necessary
```

Comment:
we can use https://github.com/voteagora/agora-next/blob/main/src/lib/serverVerifyMessage.ts#L6 for this

Reply:
multi-sigs are gonna be a pretty common use case here

Reply:
addressed this one in https://github.com/voteagora/agora-next/pull/269 by accident @stepandel